APIComparer.Website/CompareModule.cs

```diff
@@ -12,17 +14,20 @@ public class CompareModule : NancyModule
     {
         ILog logger = LogManager.GetLogger<CompareModule>();
 
-        public CompareModule(IRootPathProvider rootPathProvider, IBus bus)
+        public CompareModule(IRootPathProvider rootPathProvider, IBus bus, NuGetBrowser nuGetBrowser)
         {
-            Get["/compare/{nugetpackageid}/{leftversion:semver}...{rightversion:semver}"] = ctx =>
+            Get["/compare/{nugetpackageid}/{leftversion}...{rightversion}"] = ctx =>
             {
-                var leftVersion = SemanticVersion.Parse(ctx.leftversion);
-                var rightVersion = SemanticVersion.Parse(ctx.rightversion);
                 var nugetPackageId = (string)ctx.nugetpackageid;
 
+                var leftVersion = ExpandVersionIfNeeded(ctx.leftversion, nugetPackageId, nuGetBrowser);
+                var rightVersion = ExpandVersionIfNeeded(ctx.rightversion, nugetPackageId, nuGetBrowser);
```

Comment:
Should we response redirect instead if we where able to expand the version? (that would improve caching?)

Ie /somenuget/1.0 => redirects to /somenuget/1.0.4   (given that 1.0.4 was found when expanding)